lib/SILGen/SILGenLValue.cpp

```diff
@@ -1427,7 +1427,13 @@ namespace {
         // Create the allocating setter function. It captures the base address.
         auto setterInfo = SGF.getConstantInfo(setter);
         SILValue setterFRef = SGF.emitGlobalFunctionRef(loc, setter, setterInfo);
-        auto setterSubs = SGF.getFunction().getForwardingSubstitutionMap();
+
+        // Get the forwarding substitutions of the containing type. Since the
+        // setter cannot have additional type parameters (compared to its
+        // containing type), we can simply get the forwarding substitutions from
+        // the setter.
+        SILFunction *setterFunc = SGF.SGM.getFunction(setter, NotForDefinition);
+        auto setterSubs = setterFunc->getForwardingSubstitutionMap();
```

Comment:
You'll want to build a substitution map like the code about does for `ctor`, e.g.,

```
SubstitutionMap subs = ValType->getMemberSubstitutionMap(
                        SGF.getModule().getSwiftModule(), setter);
```

Reply:
I was thinking too complicated. This substitution map is already available as a member of the class GetterSetterComponent.

Reply:
Ah, cool!